src/winning_check.c

```diff
@@ -0,0 +1,258 @@
+                start_of_sequence[0] = carts[j + 1].value;
+                count = 1;
+            }
+        }
+    }
+    if (count == 4 && carts[6].value == 2 && carts[0].value == 14) { return 1; }// check if ace take part in set
+    return 0;
+}
+
+void winner_check(player_t *players, int numbers_of_player, card_t table_carts[5], int *how_many_players_win, player_t *table_of_winners ) {
+    int player_score[numbers_of_player];//players score
+    for (int i = 0; i < numbers_of_player; ++i) {
+        player_score[i]=0;
+    }
+    int highest_cart_value[numbers_of_player];//highest cart in players hand// include table carts important in looking for winer without any set
+    int start_of_sequence[numbers_of_player];//cart that start sequence of "strit" // looking for winner
+    int power_of_pair_carts[numbers_of_player][4];//power of cart that make sets of pair or triple
+    int highest_color_cart[numbers_of_player];//highest card in color set/ if therw is not any color - 0
+    for (int i = 0; i < numbers_of_player; ++i) {
+        for (int j = 0; j < 4; ++j) {
+            power_of_pair_carts[i][j] = 0;
+        }
+    }
+    for (int i = 0; i < numbers_of_player; ++i) {
+        card_t player_cart[7];
+        for (int j = 0; j < 5; ++j) {
+            player_cart[j] = table_carts[j];
+        }
+        player_cart[5] = players[i].cards[0];
+        player_cart[6] = players[i].cards[1];
+        buble_sort_for_carts(player_cart);
+        highest_cart_value[i] = player_cart[0].value;
+        int is_sequence = is_in_sequence(player_cart, &start_of_sequence[i]);
+        int is_color = serching_for_color(player_cart, &highest_color_cart[i]);
+        int strength_of_pair = fstrength_of_pair(player_cart, &power_of_pair_carts[i][0]);
+        if (is_color && is_sequence) {
+            player_score[i] = 8;
+            continue;
+        }
+        if (is_color) {
+            if (strength_of_pair < 4) {
+                player_score[i] = 5;
+                continue;
+            }
+
+            player_score[i] = strength_of_pair + 2;
+            continue;
+        }
+        if (is_sequence) { player_score[i] = 4; } else { player_score[i] = strength_of_pair; }
+    }
+    int count = 0;
+    int how_many_player_with_this_same_amount_of_points = 0;
+    int max = 0;
+    for (int i = 0; i < numbers_of_player ; ++i) {
+        if (player_score[i] > max) { max = player_score[i];continue; }
+        if (player_score[i] == max) {
+            count += 1;
+            if (how_many_player_with_this_same_amount_of_points < count) { how_many_player_with_this_same_amount_of_points = count; }
+        } else count = 0;
+    }
+    if (how_many_player_with_this_same_amount_of_points == 0) {
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (player_score[i] == max) {
+                table_of_winners[0] = players[i];
+                how_many_players_win[0]+=1;
```

Comment:
`how_many_players_win` is not an array use `*how_many_players_win`.
I think you ment `table_of_winners[*how_many_players_win]`